raiden/network/rpc/client.py

```diff
@@ -1088,6 +1076,11 @@ def get_block(self, block_identifier: BlockIdentifier) -> BlockData:
         """Given a block number, query the chain to get its corresponding block hash"""
         return self.web3.eth.getBlock(block_identifier)
 
+    def sync_nonce(self) -> None:
+        self._available_nonce = discover_next_available_nonce(
+            self.web3, self.eth_node, self.address
+        )
+
```

Comment:
AFAICS this is only used in the smoketest. 
Therefore I'd either make it a private method or use a custom subclass with this method in the smoketest.